Add documentation for configuring the password hashing algorithm and its properties (#7697)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/security-settings.md

- `plugins.security.check_snapshot_restore_write_privileges` (Static): Enforces write privilege evaluation when creating snapshots. Default is `true`.

If you change any of the following password hashing properties, you must rehash all internal passwords to ensure compatibility and security.
{: .warning}

- `plugins.security.password.hashing.algorithm`: (Static): Specifies the password hashing algorithm to use.

  Valid values are:
  
  - `BCrypt` (Default)
  - `PBKDF2` 

> - `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with BCrypt. Valid values are from `4` to `31`, inclusive. Default is `12`.

Reviewer: What do we mean by inclusive?

```suggestion
- `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with `BCrypt`. Valid values are a range from `4` to `31`, inclusive. Default is `12`.
```

Author: Just meant that the range includes 4 and 31 not just the numbers between them

Author: Thoughts on "Valid values range from `4` to `31`, inclusive." ?

Reviewer 2: @dancristiancecoi Alternatively, feel free to use the math notation for ranges (`[4, 31]`).

Reviewer: Updated the suggestion accordingly.

Author: Thanks a lot! Just a tiny thing @Naarcha-AWS, was that space removed on purpose?

> - `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with `BCrypt`. Valid values are from`4` to `31`, inclusive. Default is `12`.

Reviewer 3: ```suggestion
- `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with `BCrypt`. Valid values are between `4` and `31`, inclusive. Default is `12`.
```

> - `plugins.security.password.hashing.bcrypt.minor` (Static): Specifies the minor version of the BCrypt algorithm to use for password hashing.

Reviewer: ```suggestion
- `plugins.security.password.hashing.bcrypt.minor` (Static): Specifies the minor version of the `BCrypt` algorithm to use for password hashing.
```

  Valid values are:

  - `A`
  - `B`
  - `Y` (Default)

- `plugins.security.password.hashing.pbkdf2.function` (Static): Specifies the pseudo-random function applied to the password.

  Valid values are:

  - `SHA1`
  - `SHA224`
  - `SHA256` (Default)
  - `SHA384`
  - `SHA512`

> - `plugins.security.password.hashing.pbkdf2.iterations` (Static): Specifies the number of times the pseudo-random function is applied to the password. Default is `600,000`.

Reviewer 3: ```suggestion
- `plugins.security.password.hashing.pbkdf2.iterations` (Static): Specifies the number of times that the pseudo-random function is applied to the password. Default is `600,000`.
```
